rest/replicator_test.go

```diff
@@ -3810,3 +3810,186 @@ func TestActiveReplicatorPullConflictReadWriteIntlProps(t *testing.T) {
 		})
 	}
 }
+
+// This test ensures that the tombstone revision wins over non-tombstone revision
+// whilst applying default conflict resolution policy through pushAndPull replication.
+func TestDefaultConflictResolverWithTombstone(t *testing.T) {
+	if base.GTestBucketPool.NumUsableBuckets() < 2 {
+		t.Skipf("test requires at least 2 usable test buckets")
+	}
+	if !base.TestUseXattrs() {
+		t.Skip("This test only works with XATTRS enabled")
+	}
+	defer base.SetUpTestLogging(base.LevelDebug, base.KeyAll)()
+
+	// requireErrorKeyNotFound asserts that reading specified document body via SDK
+	// returns a key not found error.
+	requireErrorKeyNotFound := func(rt *RestTester, docID string) {
+		var body []byte
+		_, err := rt.Bucket().Get(docID, &body)
+		require.EqualError(t, err, fmt.Sprintf("Error during Get %s: key not found", docID))
+	}
+
+	// waitForTombstone waits until the specified tombstone revision is available
+	// in the bucket backed by the specified RestTester instance.
+	waitForTombstone := func(rt *RestTester, docID, revID string) {
+		require.NoError(t, rt.WaitForCondition(func() bool {
+			doc, _ := rt.GetDatabase().GetDocument(docID, db.DocUnmarshalAll)
+			return doc.IsDeleted() && len(doc.Body()) == 0
+		}))
+	}
+
+	// createOrUpdateDoc creates a new document or update an existing document with the
+	// specified document id, revision id and body value in a channel named "alice".
+	createOrUpdateDoc := func(rt *RestTester, docID, revID, bodyValue string) string {
+		body := fmt.Sprintf(`{"key":%q,"channels":["alice"]}`, bodyValue)
+		dbURL := "/db/" + docID
+		if docID != "" {
```

Comment:
Is this supposed to be checking revID, not docID?

Reply:
Makes sense, this is revised in latest revision.